ush/set_cycle_and_obs_timeinfo.py

```diff
@@ -624,97 +624,224 @@ def get_obs_retrieve_times_by_day(
     # Get list of field groups to be verified.
     vx_field_groups = vx_config['VX_FIELD_GROUPS']
 
-    # Define dictionary containing information about all field groups that
-    # can possibly be verified.  This information includes their temporal
-    # characteristics (cumulative vs. instantaneous) and the mapping between
-    # the observation type and the field group.
-    vx_field_info = {'cumul': [{'obtype': 'CCPA',   'field_groups': ['APCP']},
-                               {'obtype': 'NOHRSC', 'field_groups': ['ASNOW']}],
-                     'inst':  [{'obtype': 'MRMS',   'field_groups': ['REFC', 'RETOP']},
-                               {'obtype': 'NDAS',   'field_groups': ['ADPSFC', 'ADPUPA']}]
-                    }
-
-    # Keep only those items in the dictionary vx_field_info defined above that
-    # have field groups that appear in the list of field groups to verify.
-    for obs_time_type, obtypes_to_field_groups_dict_list in vx_field_info.copy().items():
-        for obtypes_to_field_groups_dict in obtypes_to_field_groups_dict_list.copy():
-            obtype = obtypes_to_field_groups_dict['obtype']
-            field_groups = obtypes_to_field_groups_dict['field_groups']
-            field_groups = [fg for fg in field_groups if fg in vx_field_groups]
-            obtypes_to_field_groups_dict['field_groups'] = field_groups
-            if not field_groups: obtypes_to_field_groups_dict_list.remove(obtypes_to_field_groups_dict)
-        if not obtypes_to_field_groups_dict_list: vx_field_info.pop(obs_time_type)
-
-    # Create dictionary containing the temporal characteristics as keys and
-    # a string list of obs types to verify as the values.
-    obs_time_type_to_obtypes_dict = dict()
-    for obs_time_type, obtypes_to_field_groups_dict_list in vx_field_info.items():
-        obtype_list = [a_dict['obtype'] for a_dict in obtypes_to_field_groups_dict_list]
-        obs_time_type_to_obtypes_dict[obs_time_type] = obtype_list
-
-    # Initialize the return variable.
-    obs_retrieve_times_by_day = dict()
-
-    # Define timedelta object representing a single day.
+    # Define a list of dictionaries containing information about all the obs
```

Comment:
This is a pretty substantial re-write after my initial review. Can you provide a brief summary on why this re-write was needed?

Reply:
@mkavulich Changes to this function are a bug fix for cumulative fields (`APCP` and `ASNOW`) to distinguish between the verification comparison times (i.e. the times at which the forecast value of say `APCP06h` is compared to its obs value) and the times at which observation files are needed (i.e. need to be retrieved/staged).

The original code was treating these as the same thing.  That is ok for non-cumulative (instantaneous) fields and field groups (`REFC`, `RETOP`, `SFC`, and `UPA`) but not for the cumulative ones.  For example, if we're verifying APCP06h for a 00Z forecast, the vx comparison times are 06Z, 12Z, 18Z, etc, but the times at which the (hourly) CCPA obs files are needed are 01Z, 02Z, 03Z, etc because the observed APCP06h need to be constructed from the 1-hour obs.  The new code calculates the obs retrieval times correctly for cumulative fields, which is then used in the `get_obs()` function to look for and possibly retrieve the necessary obs files.

This bug showed up when I tried using a forecast output interval (`VX_FCST_OUTPUT_INTVL_HRS`) that is greater than 1, in particular when I switched to using 6-hourly forecast output from the AI/ML models.  I don't remember exactly how that relates to the obs retrieval times, but somehow they were related, and I had to make these changes to get the AI/ML WE2E tests to work.